agent/src/beerocks/fronthaul_manager/ap_manager/ap_manager_thread.cpp

```diff
@@ -209,7 +209,7 @@ void ap_manager_thread::ap_manager_fsm()
         // External events
         int ext_events_fd = ap_wlan_hal->get_ext_events_fd();
         if (ext_events_fd > 0) {
-            ap_hal_ext_events = new Socket(ext_events_fd);
+            ap_hal_ext_events = Socket::socketFactory(ext_events_fd);
```

Comment:
(nitpick) It would be better to have a separate commit for creating the factor method, and a different one for adding the DefaultSocket class.